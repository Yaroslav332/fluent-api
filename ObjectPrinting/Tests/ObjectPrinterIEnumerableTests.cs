using System;
using System.Globalization;
using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;

namespace ObjectPrinting.Tests
{
    [UseReporter(typeof(DiffReporter))]
    [TestFixture]
    public class ObjectPrinterIEnumerableTests
    {
        [Test]
        public void Print_Should_CorrectlyDisplayIEnumerables()
        {
            var r = new LibraryRecords()
            {
                ActiveUsers = {"Alex", "Mary", "Stan"},
                ReadersVisits = {{"Alex", new DateTime(1, 1, 1)}, {"Winona", new DateTime(3, 3, 3)}}
            };
            var printer = ObjectPrinter.For<LibraryRecords>();
            Approvals.Verify(printer.PrintToString(r));
        }
        
        [Test]
        public void SetCulture_ShouldWork_ForCollectionsElements()
        {
            var r = new LibraryRecords()
            {
                ActiveUsers = {"Alex", "Mary", "Stan"},
                ReadersVisits = {{"Alex", new DateTime(1, 1, 1)}, {"Winona", new DateTime(3, 3, 3)}}
            };
            var printer = ObjectPrinter.For<LibraryRecords>()
                .SetCulture(new CultureInfo("en"));
            Approvals.Verify(printer.PrintToString(r));
        }
        
        [Test]
        public void SetSerializeMethodForType_ShouldWork_ForCollectionsElements()
        {
            var r = new LibraryRecords()
            {
                ActiveUsers = {"Alex", "Mary", "Stan"},
                ReadersVisits = {{"Alex", new DateTime(1, 1, 1)}, {"Winona", new DateTime(3, 3, 3)}}
            };
            var printer = ObjectPrinter.For<LibraryRecords>()
                .SetSerializeMethodForType<string>(s => s + "123");
            Approvals.Verify(printer.PrintToString(r));
        }
        
        [Test]
        public void ExcludeType_ShouldExclude_CollectionsElements()
        {
            var r = new LibraryRecords()
            {
                ActiveUsers = {"Alex", "Mary", "Stan"},
                ReadersVisits = {{"Alex", new DateTime(1, 1, 1)}, {"Winona", new DateTime(3, 3, 3)}}
            };
            var printer = ObjectPrinter.For<LibraryRecords>()
                .ExcludeType<DateTime>();
            Approvals.Verify(printer.PrintToString(r));
        }
        
        [Test]
        public void Print_ShouldSerializeAsEmpty_EmptyIEnumerable_Or_WithExcludedType()
        {
            var r = new LibraryRecords()
            {
                ReadersVisits = {{"Alex", new DateTime(1, 1, 1)}, {"Winona", new DateTime(3, 3, 3)}}
            };
            var printer = ObjectPrinter.For<LibraryRecords>()
                .ExcludeType<string>();
            Approvals.Verify(printer.PrintToString(r));
        }
        
    }
}