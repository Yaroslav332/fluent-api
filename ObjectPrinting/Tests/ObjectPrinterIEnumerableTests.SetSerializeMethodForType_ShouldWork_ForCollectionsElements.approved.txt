LibraryRecords
	ActiveUsers = [
		Alex123,
		Mary123,
		Stan123
		]
	ReadersVisits = [
		KeyValuePair`2
			Key = Alex123
			Value = 01.01.0001 0:00:00,
		KeyValuePair`2
			Key = Winona123
			Value = 03.03.0003 0:00:00
		]
	BooksCollection = [
		A123,
		B123,
		C123,
		D123,
		E123
		]
